### DISCUSSION TITLE
Mail profiles for ios thunderbird etc?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [docker-mailserver/docker-mailserver#3260] [Q&A]
Setting up mail on iphone is always such a pain. I could not find but might missed it is there some way to generate a mail profile file that you can load on iphone or mail client to setup the mail account for you?

**Comment by georglauterbach** [comment]
There is no such thing (as of now) for DMS AFAIK.

**Reply by Jarrodsz** [reply]
Not really a dealbreaker but setting up mail in different clients like thunderbird and apple mail is a bit cybersome sometimes, luckily it only has to be done once. I know some mailservers provide a mail_profile that can be downloaded and loads all the correct settings. 
Perhaps something for the future. It would help for non technical users, say I create a mailbox for someone i can just share the profile with them and they can install it without me having to do it.

**Comment by casperklein** [comment]
Maybe [this](https://docker-mailserver.github.io/docker-mailserver/latest/config/best-practices/autodiscover/) helps.

**Reply by Jarrodsz** [reply]
I think this provides indeed such profile solution. Seems overly complicated though? Would require a docker compose with manual entry for each mail domain. Needs ldap? Scsnning the docs. 

Way to much hassle and it would not integrate well with docker mailserver. From scanning the docs one would need to manually maintain this, you would end up with docker mailserver and a seperate docker conpose for this service just not dry. Its not a big deal but sadly iphone does not sync mail settings from osx. So stupid also apple mail config is terrible. Then i use thunderbird for auto replies whoch needs again totally different setup.

i guess for now its just better to manually set mail accounts on clients than using this solution. At least you only have to set them up twice for iphone and apple mail and in my vase thunderbird it sucks but better then this solution.

If docker mailserver would integrate just the profile file generation that would be cool could save a lot of time but not a dealbraker its already aeesome as-is.

**Comment by Jarrodsz** [comment]
We discussed this before.
The external plugin seems messy implementation and not a nice way of doing it.

Would it be possible to write a small plugin that creates the profiles instead?

I guess the profile is just a file with some standarized format so it should be possible to create a profile file for mail clients based on all content that is in the mailserver docker?

The reason for adking this again its really cybersome and buggy on osx and ios mail clients any mail client actually to setup mail sever settings.

Thus it would be very beneficial to have the option to generate profile files with setup.sh that would output to the host running the docker container.

It would then be easy to download those files and use them in your mail clients thus reducing thr time to setup new mail accounts.

I guess with the help of gpt4 this could be done in 1 hour.

Is there interest in this ?

If i where to create a plugin Pr for this where would i start does docker mailserver has a plugin system?

I guess it could be even as simple as a shell script that generates these profiles fetching data from the container and output to several profiles for each account in a loop!

Would love to learn more and if anyone is interested in working on this

**Reply by georglauterbach** [reply]
You can place a script under target/bin and then use target/bin/setup to call it.

**Comment by casperklein** [comment]
Is "mail profile file" mac/apple specific? Or is this a common standard?

**Comment by Jarrodsz** [comment]
Common mail profiles for clients

**Comment by Jarrodsz** [comment]
A quick draft:

```bash
#!/bin/bash

# Mail server settings
MAIL_SERVER="mail.example.com"
IMAP_PORT="993"
SMTP_PORT="587"
USERNAME="your_username"
PASSWORD="your_password"

# Define the profile XML template function
generate_profile_template() {
    cat <<EOF
<?xml version="1.0" encoding="UTF-8"?>
<root>
    <IncomingServer>
        <Hostname>${MAIL_SERVER}</Hostname>
        <Port>${IMAP_PORT}</Port>
        <Username>${USERNAME}</Username>
        <Password>${PASSWORD}</Password>
        <UseSSL>true</UseSSL>
        <Authentication>password</Authentication>
    </IncomingServer>
    <OutgoingServer>
        <Hostname>${MAIL_SERVER}</Hostname>
        <Port>${SMTP_PORT}</Port>
        <Username>${USERNAME}</Username>
        <Password>${PASSWORD}</Password>
        <UseSSL>true</UseSSL>
        <Authentication>password</Authentication>
        <AddPortToURL>true</AddPortToURL>
        <UseSPA>false</UseSPA>
    </OutgoingServer>
</root>
EOF
}

# Create profile files function
create_profile() {
    local device="$1"
    local profile_file="mail_profile_${device}.xml"
    
    generate_profile_template > "${profile_file}"
    echo "Created profile file for ${device}: ${profile_file}"
}

# Call create_profile function for different clients
clients=("macOS" "iOS" "Thunderbird")

for client in "${clients[@]}"; do
    create_profile "${client}"
done

echo "Mail profile files generated successfully."
```

**Reply by casperklein** [reply]
Looks good, but what's the point in creating three identically files?

**Reply by Jarrodsz** [reply]
Well not it was just a quick gpt illustration. Is there interest in integrating this? I could dive a bit more into it and enhanche it to work for production / test. And create a Pr for this?

**Reply by casperklein** [reply]
I don't know if there is great demand for it. You can check existing issues, if someone else requested this. Independent of that, if it's a small addition that just works, I don't see a reason not to merge it.

There are a few questions that came to my mind, that would have to be solved:

1. Should the files be created within the container or outside?
2. How do you get the files from the container or host to the clients?
3. Are the credentials manually added or in some automated way? For the second, the container does not know the plaintext passwords.

Not sure if you understood my previous questions when you answered it: Are those mail profile files also suitable for other non apple mail clients? I ask that, because I couldn't find any information about this "mail profile" files. Is there an RFC or other document for that? You mentioned Thunderbird, but I couldn't find information wether it's supported or not.

**Reply by Jarrodsz** [reply]
> I don't know if there is great demand for it. You can check existing issues, if someone else requested this. Independent of that, if it's a small addition that just works, I don't see a reason not to merge it.
> 
> 
> 
> There are a few questions that came to my mind, that would have to be solved:
> 
> 
> 
> 1. Should the files be created within the container or outside?
> 

The setup.sh profilegen command would gather data from the container the outputs to the host os for easy copy paste.

> 2. How do you get the files from the container or host to the clients?

You have to download then 
> 
> 3. Are the credentials manually added or in some automated way? For the second, the container does not know the plaintext passwords.
The credentials would be gathered from the docker container. 

Did not thought about the credentials beeing not included. I guess then the passwords are left out also adds safety. The mail profile contains only the server connection settings and you add the pasword as final step in the mail client after the mail profile file has been importered.
I keep passwords in 1 password offline anyway assume this is the general way of storing passwords
> 
> 
> 
> Not sure if you understood my previous questions when you answered it: Are those mail profile files also suitable for other non apple mail clients? I ask that, because I couldn't find any information about this "mail profile" files. Is there an RFC or other document for that? You mentioned Thunderbird, but I couldn't find information wether it's supported or not.

Yes its possible for each client to have a mail profile file im not  sure there is a generic standard for server settings for each 
client could be that it had to generate a different file for ios apple mail thunderbird etc.

Gpt says 

As of my last update in September 2021, there isn't a universally accepted standard RFC (Request For Comments) or file format specifically for exporting/importing mail client server settings across all mail clients.

Most email clients have their own proprietary ways of exporting or importing configurations. Some might support importing from certain other popular clients, but there's no one-size-fits-all format. For instance, Microsoft Outlook uses ".pst" and ".ost" files, Mozilla Thunderbird uses a proprietary profile folder structure, and so on.

Would just create a profile for each client in the script
> 
> 
> 
>

**Comment by Jarrodsz** [comment]
Its less invase and does not include polluting dns with entries like this

https://hub.docker.com/r/monogramm/autodiscover-email-settings/

We would only need to find out the rfc for each client then modify the bash script to export the files.

We could even split this task up so a few users test and generate a function for each client. Could be do anle in 1/2 hours.

Then after that never any manual fiddling setting up mail client settings with there quircks just import the mail profile and add a password and your set.

Huge timesaver in my opinion and a great additon to docker mailsevers ease of use!

Hope there is interest in this maybe thr core team can chime in and we can implement this 

Since the mail client profile rfcs are fixed i think this is a one time endavour.

Probably start with:
- Thunderbird
- Apple mail
- Ios
- Microsoft outlook

**Reply by casperklein** [reply]
Now I start to understand. I thought there is a global "profile file" standard. So you want to create different files (contents) for each client. This seems pretty work intensive. Also, how would you proper test this, e.g. when a new client version is released. You'll have to verify that it still works.

For me, the autodiscover solution seems to be the easiest one. Afaik it's universal and supported by all major clients. From an end-user perspective, no file have to be downloaded / imported. Just entering the email address + password.

BTW: Why are you thinking adding DNS entries is pollution? What's wrong with adding those? An end-user will never see them, therefore can't be confused by them.

That said, if you want it, go for it 👍 We'll review the PR and if it's sane - merge it.
